src/org/labkey/test/tests/DomainFieldTypeChangeTest.java

```diff
@@ -0,0 +1,205 @@
+        dgen.createDomain(createDefaultConnection(), "IntList", Map.of("keyName", "id"));
+
+        log("Inserting sample rows in the list");
+        dgen.addCustomRow(Map.of("name", "first", "testInteger", "1",
+                "testDecimal", "1.10", "testDate", "01-01-2022",
+                "testBoolean", "true"));
+        dgen.addCustomRow(Map.of("name", "Second", "testInteger", "2",
+                "testDecimal", "2.20", "testDate", "01-02-2022",
+                "testBoolean", "false"));
+        dgen.addCustomRow(Map.of("name", "Third", "testInteger", "3",
+                "testDecimal", "3.30", "testDate", "01-03-2022",
+                "testBoolean", "true"));
+        dgen.insertRows(createDefaultConnection(), dgen.getRows());
+
+        log("Verifying Integer to Decimal change");
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", listName);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldsPanel();
+        domainFormPanel.getField("testInteger").setType(FieldDefinition.ColumnType.Decimal);
+        domainFormPanel.getField("testBoolean").setNumberFormat("yes;no");
+        domainDesignerPage.clickFinish();
+
+        clickAndWait(Locator.linkWithText(listName));
+        DataRegionTable table = new DataRegionTable("query", getDriver());
+        checker().verifyEquals("Incorrect values after changing integer to decimal", Arrays.asList("1.0", "2.0", "3.0"),
+                table.getColumnDataAsText("testInteger"));
+
+        log("Verifying changin data fields to string");
+        domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", listName);
+        domainFormPanel = domainDesignerPage.fieldsPanel();
+        domainFormPanel.getField("testInteger").setType(FieldDefinition.ColumnType.String);
+        domainFormPanel.getField("testDecimal").setType(FieldDefinition.ColumnType.String);
+        domainFormPanel.getField("testDate").setType(FieldDefinition.ColumnType.String);
+        domainFormPanel.getField("testBoolean").setType(FieldDefinition.ColumnType.String);
+        domainDesignerPage.clickFinish();
+
+        clickAndWait(Locator.linkWithText(listName));
+        table = new DataRegionTable("query", getDriver());
+        checker().verifyEquals("Incorrect values after changing integer to string", Arrays.asList("1", "2", "3"),
+                table.getColumnDataAsText("testInteger"));
+        checker().verifyEquals("Incorrect values after changing decimal to string", Arrays.asList("1.1", "2.2", "3.3"),
+                table.getColumnDataAsText("testDecimal"));
+        checker().verifyEquals("Incorrect values after changing date to string", Arrays.asList("2022-01-01 00:00:00", "2022-01-02 00:00:00", "2022-01-03 00:00:00"),
+                table.getColumnDataAsText("testDate"));
+        checker().verifyEquals("Incorrect values after changing boolean to string", Arrays.asList("yes", "no", "yes"),
```

Comment:
To confirm that the tyoe migration was successful, I recommend also doing an update of one of these fields to enter a value that wasn't previously supported, like "1Test" for testInteger.